docs/development/functional_architecture.md

```diff
@@ -0,0 +1,56 @@
+# Parsec client functional components
+
+Note: This is a work in progress.
+
+- [Parsec client functional components](#parsec-client-functional-components)
+  - [Mountpoint](#mountpoint)
+  - [Gui](#gui)
+  - [Core](#core)
+  - [Metadata server](#metadata-server)
+
+## Mountpoint
+
+The `mountpoint` component provide a virtual filesystem that allow other applications to access its contents.
+This is done using the os filesystem's API that follow native request to the component.
+
+## Gui
+
+The `GUI` component provide to the user a graphical interface and interact directly with the `core` or the os filesystem (but never with the `mountpoint`).
+On contrary to the `mountpoint` component, the `gui` subscribe to events provided by the `metadata server` to notify the user in case of concurrent operation, deletion of a group, etc.
+
+## Core
+
+On top of the 2 previous componenet, the `core` component is responsible for the client side logic and his composed of 5 sub-modules (see figure below):
+
+![Core submodules](../figures/parsec_core_submodules.png)
+
+- The `Virtual File System` (VFS) (`Système de Fichiers Virtuels`): Receive request related to the filesystem originating from the `mountpoint`, thus expose a filesystem like api.
+  To optimize its performance, that component don't push the modifications to the `metadata server`, it only store the encrypted modifications locally on the machine.
```

Comment:
Why keeping the French translation?
```suggestion
- The `Virtual File System` (VFS): Receive filesystem-related requests originating from the `mountpoint`, exposing a filesystem like API.
  To optimize its performance, this component doesn't push the changes to the `metadata server`, instead it only stores the encrypted changes locally on the machine.
```

Reply:
The used schema use french word, that the way I find to do a mapping

Reply:
done